tgui/packages/tgui/components/TextArea.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * @file
+ * @copyright 2020 Aleksej Komarov
+ * @author Original Aleksej Komarov
+ * @author Changes Warlockd (https://github.com/warlockd)
+ * @license MIT
+ */
+
+
+import { classes, isFalsy } from 'common/react';
+import { Component, createRef } from 'inferno';
+import { Box } from './Box';
+
+
+const toInputValue = value => {
+  if (isFalsy(value)) {
+    return '';
+  }
+  return value;
+};
+
+export class TextArea extends Component {
+  constructor(props, context) {
+    super(props, context);
+    this.textareaRef = createRef();
+    this.fillerRef = createRef();
+    const {
+      autoresize = false,
+      dontUseTabForIndent = false,
+      value,
+    } = props;
+    this.state = {
+      value: value,
```

Comment:
This is the typical port of TextArea from /tg/ but includes this value state.
This helped with the jank that i got when editing the field and having it snap back and forth due to the value being editing and set at the same time.

Keeping it as internal state fixed that.